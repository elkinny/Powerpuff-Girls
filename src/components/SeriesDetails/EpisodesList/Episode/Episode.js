import React from 'react';
import Link from 'next/link';
import PropTypes from 'prop-types';

import style from './style.module.scss';

const Episode = ({
  id, season, number, name, image,
}) => (
  <Link href="/[episodeId]" as={`/${id}`}>
    <div className={style.card} style={{ backgroundImage: `url(${image})` }}>
      <div className={style.wrapper}>
        <span className={style.name}>{name}</span>
      </div>
      <span className={style.number}>
        {season}
        .
        {number}
      </span>
    </div>
  </Link>
);

Episode.propTypes = {
  id: PropTypes.number.isRequired,
  season: PropTypes.number.isRequired,
  number: PropTypes.number.isRequired,
  name: PropTypes.string.isRequired,
  image: PropTypes.string.isRequired,
};

export default Episode;
