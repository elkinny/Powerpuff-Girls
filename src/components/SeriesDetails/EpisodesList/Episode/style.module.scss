@import '../../../../styles/constants';

.card {
  width: 300px;
  height: 200px;
  min-width: 300px;
  background-size: cover;
  background-position: center; 
  position: relative;
  margin: 20px;

  &:hover {
    cursor: pointer;

    .wrapper {
      opacity: 1;
    }

    .number {
      background-color: rgba(0,0,0,0);
    }
  }

  .number {
    font-size: 15px;
    position: absolute;
    background-color: rgba(0,0,0,0.6);
    padding: 5px 11px;
    top: 0;
    right: 0;
    transition: background-color 0.3s;
  }
    
  .wrapper {
    position: absolute;
    background-color: rgba(0,0,0,0.8);
    height: 100%;
    width: 100%;
    padding: 0 10px;
    display: flex;
    opacity: 0;
    transition: opacity 0.3s;

    .name {
      font-size: 24px;
      margin: auto;
      text-align: center;
    }
  }
}