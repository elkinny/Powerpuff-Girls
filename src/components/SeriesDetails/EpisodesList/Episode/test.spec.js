import React from 'react';
import { render } from '@testing-library/react';

import Episode from './Episode';

describe('Episode', () => {
  test('renders Episode component', () => {
    const { container } = render(<Episode
      id={367}
      name="Episode Name"
      season={1}
      number={4}
      image="/placeholder.png"
    />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
