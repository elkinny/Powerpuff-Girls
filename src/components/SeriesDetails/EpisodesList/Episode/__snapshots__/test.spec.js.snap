// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Episode renders Episode component 1`] = `
<div
  style="background-image: url(/placeholder.png);"
>
  <div>
    <span>
      Episode Name
    </span>
  </div>
  <span>
    1
    .
    4
  </span>
</div>
`;
